template-for-repository/proofs/Makefile.common

```diff
@@ -109,8 +109,13 @@ PROOFDIR ?= $(abspath .)
 CBMCDIR ?= $(PROOF_ROOT)/cbmc
 
 # Default CBMC flags used for property checking and coverage checking
+CBMCFLAGS += --malloc-fail-null
+CBMCFLAGS += --malloc-may-fail
 CBMCFLAGS += --unwind 1 $(CBMC_UNWINDSET)
 
+# Remove DISABLED_FLAGS from CBMCFLAGS
+CBMCFLAGS := $(filter-out $(DISABLED_FLAGS),$(CBMCFLAGS))
```

Comment:
Your method, however, works only for flags without arguments and fails to work for flags with arguments.  Suppose I want to remove --unwind 1.

One approached three of us discussed off line was to define variables

MALLOC_CAN_FAIL=--malloc-fail-null
MALLOC_MAY_FAIL=--malloc-may-fail
POINTER_PRIMITIVE_CHECK=--pointer-primitive-check

CBMCFLAGS += $(MALLOC_CAN_FAIL)
CBMCFLAGS += $(MALLOC_MAY_FAIL)
CBMCFLAGS += $(POINTER_PRIMITIVE_CHECK)

This would allow an individual proof not wanting the --pointer-primitive-check to write a proof Makefile to include

undefine POINTER_PRIMITIVE_CHECK

See the [Undefine Directive](https://www.gnu.org/software/make/manual/html_node/Undefine-Directive.html).  

I recommend experimenting with this and rewriting Makefile.common in this style.  If it works, it would be a general solution doing a general service to the project.

Reply:
This won't work because the proof-specific Makefile is read before Makefile.common. At that point, there is no variable to undefine.

Suggestion: do this inside Makefile.common
```Makefile
MALLOC_CAN_FAIL ?= --malloc-fail-null
```

which will set `MALLOC_CAN_FAIL` iff it hasn't already been set. Folks who want to override it in a proof can do
```Makefile
MALLOC_CAN_FAIL := 
```
inside the proof Makefile. Furthermore, if people would like to disable a flag for the entire repo, they could do the same thing inside Makefile-project-defines.

So I suggest to put these `?=` definitions *after* including Makefile-project-defines, so that proof authors have a chance to disable them both per-proof and project-wide.

Reply:
Makes sense, thanks! I will make the change.

Reply:
One suggestion: use a standard naming scheme. `CBMC_FLAG_MALLOC_CAN_FAIL`